drivers/dai/intel/ssp/ssp.c

```diff
@@ -1586,14 +1687,37 @@ static int dai_ssp_set_config_blob(struct dai_intel_ssp *dp, const struct dai_co
 
 	ssp->state[DAI_DIR_PLAYBACK] = DAI_STATE_PRE_RUNNING;
 	ssp->state[DAI_DIR_CAPTURE] = DAI_STATE_PRE_RUNNING;
+}
 
-	/* ssp blob is set by pcm_hw_params for ipc4 stream, so enable
-	 * mclk and bclk at this time.
-	 */
-	dai_ssp_mn_set_mclk_blob(dp, blob->i2s_driver_config.mclk_config.mdivc,
-				 blob->i2s_driver_config.mclk_config.mdivr);
-	ssp->clk_active |= SSP_CLK_MCLK_ES_REQ;
+static int dai_ssp_set_config_blob(struct dai_intel_ssp *dp, const struct dai_config *cfg,
+				   const void *spec_config)
+{
+	const struct dai_intel_ipc4_ssp_configuration_blob_ver_1_5 *blob15 = spec_config;
+	const struct dai_intel_ipc4_ssp_configuration_blob *blob = spec_config;
+	struct dai_intel_ssp_pdata *ssp = dai_get_drvdata(dp);
+	int err;
 
+	/* set config only once for playback or capture */
+	if (ssp->state[DAI_DIR_PLAYBACK] > DAI_STATE_READY ||
+	    ssp->state[DAI_DIR_CAPTURE] > DAI_STATE_READY)
+		return 0;
+
+	if (blob15->version == SSP_BLOB_VER_1_5) {
+		err = dai_ssp_parse_aux_data(dp, spec_config);
```

Comment:
this should be parsed before reg config?

Reply:
is it after or before doesn't too much matter (I think), if it doesn't exists, the parsing will continue to regs.